src/hawkbit-client.c

```diff
@@ -135,7 +145,7 @@ static gboolean get_binary(const gchar* download_url, const gchar* file, gint64
         CURL *curl = curl_easy_init();
         if (!curl) {
                 fclose(fp);
-                g_set_error(error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                g_set_error(error, RHU_HAWKBIT_CLIENT_CURL_ERROR, -1,
```

Comment:
The icing on the cake would be to define an error code for this like `RHU_HAWKBIT_CLIENT_ERROR_INIT`.

Reply:
Hm, this error occurs only if `curl_easy_init()` returned NULL. Isn't that a curl error?

Reply:
I was referring to something like

``` c
#define  RHU_HAWKBIT_CLIENT_CURL_ERROR_INIT -1
```

and

```suggestion
                g_set_error(error, RHU_HAWKBIT_CLIENT_CURL_ERROR, RHU_HAWKBIT_CLIENT_CURL_ERROR_INIT,
```

Reply:
I see, I've used `CURLE_FAILED_INIT` now instead.